src/pages/HomePage.tsx

```diff
@@ -34,6 +34,10 @@ const HomePage: FC<IProps & RouteComponentProps<IRouteProps>> = ({ match }) => {
   const initialId = store?.newsFlashCollection[0]?.id;
   const id = match.params.id ? parseInt(match.params.id) : initialId;
```

Comment:
This is a logical mistake. if you'll set `id` to be `initialId` when param is not provided, it will never redirect!

Reply:
You can remove the `const id` and do:
```
  if (!match.params.id) {
    return <Redirect to={`/newsflash/${initialId}`} />;
  }
```

Reply:
The last suggested solution redirects to ...newsflsah/undefined
Due to the first time the app is loaded -initialId is undefined

I would like to discuss this issue  .